src/VisualStudio/Core/Def/DocumentOutline/DocumentSymbolViewModel.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Imaging.Interop;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.VisualStudio.LanguageServices
+{
+    using SymbolKind = LanguageServer.Protocol.SymbolKind;
+
+    internal class DocumentSymbolViewModel : INotifyPropertyChanged
+    {
+        private bool _isSelected;
+        private bool _isExpanded;
+
+        public string Name { get; }
+
+        public List<DocumentSymbolViewModel> Children { get; set; }
+
+        public int StartLine { get; }
+        public int StartChar { get; }
+        public int EndLine { get; }
+        public int EndChar { get; }
+
+        public SymbolKind SymbolKind { get; }
+        public ImageMoniker ImgMoniker { get; }
+
+        public bool IsExpanded
+        {
+            get => _isExpanded;
+            set
+            {
+                if (_isExpanded != value)
+                {
+                    _isExpanded = value;
+                    NotifyPropertyChanged();
+                }
+            }
+        }
+        public bool IsSelected
+        {
+            get => _isSelected;
+            set
+            {
+                if (_isSelected != value)
+                {
+                    _isSelected = value;
+                    NotifyPropertyChanged();
+                }
+            }
+        }
+
+        public DocumentSymbolViewModel(DocumentSymbol documentSymbol)
+        {
+            this.Name = documentSymbol.Name;
+            this.Children = new List<DocumentSymbolViewModel>();
+            this.SymbolKind = documentSymbol.Kind;
+            this.ImgMoniker = GetImageMoniker(documentSymbol.Kind);
+            this.IsExpanded = true;
+            this.IsSelected = false;
+            this.StartLine = documentSymbol.Range.Start.Line;
+            this.StartChar = documentSymbol.Range.Start.Character;
+            this.EndLine = documentSymbol.Range.End.Line;
+            this.EndChar = documentSymbol.Range.End.Character;
+        }
+
+        private void NotifyPropertyChanged([CallerMemberName] string? propertyName = null)
+            => PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+
+        public event PropertyChangedEventHandler? PropertyChanged;
+
+        private static ImageMoniker GetImageMoniker(SymbolKind symbolKind)
```

Comment:
Are we able to reuse the existing code we have for getting monikers for symbols?

Reply:
Spoke to Jason - resolving since this code doesn't exist